Release_Process.md

```diff
@@ -0,0 +1,23 @@
+# Our Release Process
+
+This is a summary of the steps we go through to release a new version of BigchainDB Server.
+
+1. Update the `CHANGELOG.md` file
+2. Update `bigchaindb/version.py`
+3. Go to the [bigchaindb/bigchaindb Releases page on GitHub](https://github.com/bigchaindb/bigchaindb/releases)
+   and click the "Draft a new release" button
+4. Name the tag something like v0.7.0
```

Comment:
I'd add a pointer to http://semver.org/ somewhere here, so the person who is making the release knows that we use the `MAJOR.MINOR.PATCH` schema.